File: dlna-dmc/src/main/java/com/android/cast/dlna/OnDeviceRegistryListener.java
package com.android.cast.dlna;

import com.android.cast.dlna.device.CastDevice;

/**
 * this listener call in UI thread.
 */
public interface OnDeviceRegistryListener {
    void onDeviceAdded(CastDevice device);

    void onDeviceUpdated(CastDevice device);

    void onDeviceRemoved(CastDevice device);
}

File: dlna-dmc/src/main/java/com/android/cast/dlna/dmc/OnDeviceRegistryListener.kt
package com.android.cast.dlna.dmc

import org.fourthline.cling.model.meta.Device

/**
 * this listener call in UI thread.
 */
interface OnDeviceRegistryListener {
    fun onDeviceAdded(device: Device<*, *, *>) {}
    fun onDeviceUpdated(device: Device<*, *, *>) {}
    fun onDeviceRemoved(device: Device<*, *, *>) {}
    fun onDeviceListChanged(list: List<Device<*, *, *>>) {}
}